internal/plugin/loopback/testing_grpc_stream_test.go

```diff
@@ -16,195 +18,339 @@ import (
+		err := stream.client.CloseSend()
+		require.NoError(err)
+		require.True(stream.streamClosed)
+		_, ok := <-stream.messages
+		require.False(ok)
+		require.Empty(stream.messages)
+	})
+
+	t.Run("closeSend returns a stream is closed error", func(t *testing.T) {
+		stream := newGetObjectStream()
+		require.NotNil(stream)
+		require.NotNil(stream.client)
+
+		stream.Close()
+		require.True(stream.streamClosed)
+
+		// Validate CloseSend does return an error when the channel is closed
+		// Validate streamClosed is set to true
+		// Validate the channel messages is closed
+		err := stream.client.CloseSend()
 		require.Error(err)
+		require.Contains(err.Error(), "stream is closed")
+		require.True(stream.streamClosed)
+		_, ok := <-stream.messages
+		require.False(ok)
+		require.Empty(stream.messages)
+	})
+
+	t.Run("recv returns EOF error on closed channel", func(t *testing.T) {
+		stream := newGetObjectStream()
+		require.NotNil(stream)
+		require.NotNil(stream.client)
+
+		stream.Close()
+		require.True(stream.streamClosed)
+
+		// Validate Recv returns EOF error when the channel is closed
+		resp, err := stream.client.Recv()
+		require.Equal(io.EOF, err)
 		require.Nil(resp)
-	}()
-	wg.Wait()
-
-	// Validate CloseSend does not return an error
-	// Validate streamClosed is set to true
-	// Validate the channel messages is closed
-	var ok bool
-	err = stream.client.CloseSend()
-	require.NoError(err)
-	require.True(stream.streamClosed)
-	_, ok = <-stream.messages
-	require.False(ok)
-
-	// Validate CloseSend does return an error when the channel is closed
-	// Validate streamClosed is set to true
-	// Validate the channel messages is closed
-	err = stream.client.CloseSend()
-	require.Error(err)
-	require.Contains(err.Error(), "stream is closed")
-	require.True(stream.streamClosed)
-	_, ok = <-stream.messages
-	require.False(ok)
-
-	// Validate Recv returns EOF error when the channel is closed
-	resp, err := stream.client.Recv()
-	require.Equal(io.EOF, err)
-	require.Nil(resp)
+		require.Empty(stream.messages)
+	})
 }
 
 func Test_GetObjectStream_Server(t *testing.T) {
 	require := tr.New(t)
 
-	stream := newGetObjectStream()
-	require.NotNil(stream)
-	require.NotNil(stream.server)
+	t.Run("set header", func(t *testing.T) {
+		stream := newGetObjectStream()
+		defer stream.Close()
+		require.NotNil(stream)
+		require.NotNil(stream.server)
 
-	// Validate SetHeader does not return an error
-	err := stream.server.SetHeader(nil)
-	require.NoError(err)
-
-	// Validate SendHeader does not return an error
-	err = stream.server.SendHeader(nil)
-	require.NoError(err)
-
-	// Validate Context returns a context
-	ctx := stream.server.Context()
-	require.NotNil(ctx)
-
-	// Validate RecvMsg does not read from the channel messages
-	streamSize := len(stream.messages)
-	var msg *getObjectStreamResponse
-	err = stream.server.RecvMsg(msg)
-	require.NoError(err)
-	require.Nil(msg)
-	require.Equal(streamSize, len(stream.messages))
```

Comment:
I'm not seeing a corresponding test for this in the new `Test_GetObjectStream_Server`? Is it needed? (Disclaimer: I don't have a lot of context on what this is testing 😅 )

Reply:
The corresponding test for the server side is `recieve empty message`